packages/editor/src/components/post-publish-button/index.js

```diff
@@ -102,20 +140,32 @@ export class PostPublishButton extends Component {
 			className: 'editor-post-publish-panel__toggle',
 			isBusy: isSaving && isPublished,
 			isPrimary: true,
-			onClick: onClickToggle,
+			onClick: this.createOnClick( onClickToggle ),
 		};
 
 		const toggleChildren = isBeingScheduled ? __( 'Schedule…' ) : __( 'Publish…' );
-		const buttonChildren = <PublishButtonLabel forceIsSaving={ forceIsSaving } />;
+		const buttonChildren = (
+			<PublishButtonLabel
+				forceIsSaving={ forceIsSaving }
+				hasNonPostEntityChanges={ hasNonPostEntityChanges }
+			/>
+		);
 
 		const componentProps = isToggle ? toggleProps : buttonProps;
 		const componentChildren = isToggle ? toggleChildren : buttonChildren;
 		return (
 			<div>
+				<EntitiesSavedStates
+					isOpen={ Boolean( entitiesSavedStatesCallback ) }
+					onRequestClose={ this.closeEntitiesSavedStates }
+				/>
 				<Button
 					isLarge
 					ref={ this.buttonNode }
 					{ ...componentProps }
+					className={ classnames( componentProps.className, {
+						'editor-post-publish-button__has-changes-dot': hasNonPostEntityChanges,
```

Comment:
This reads like it should be a modifier class to me:

```suggestion
						'has-changes-dot': hasNonPostEntityChanges,
```

Which could then be referenced as:

```scss
.editor-post-publish-button .components-button.has-changes-dot { /* ... */ }
```

Or as a modifier on the publish button itself:

```scss
.editor-post-publish-button.has-changes-dot .components-button { /* ... */ }
```

Or at the very least as a modifier of a class specific to PostPublishButton, assigned to the button:

```suggestion
						'editor-post-publish-button__button': true,
						'has-changes-dot': hasNonPostEntityChanges,
```

(instead of `true` value assignment, can be a separate argument of `classnames`, outside the object)

```scss
.editor-post-publish-button__button.has-changes-dot .components-button { /* ... */ }
```

Reference: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/coding-guidelines.md#naming

Reply:
>Or at the very least as a modifier of a class specific to PostPublishButton, assigned to the button:

This makes the most sense.

It's too specific to be a modifier of `.components-button` and it's not modifying all of `.editor-post-publish-button`.